File: KonorkestraCLI/KonorkestraCLI.py
import argparse
import json
import requests
import os

# Function to read the configuration file
def read_config_file(file_path):
    if not os.path.exists(file_path):
        print(f"Error: The file {file_path} does not exist.")
        return None

    # Read and parse the JSON configuration file
    try:
        with open(file_path, 'r') as file:
            config_data = json.load(file)
            return config_data
    except json.JSONDecodeError as e:
        print(f"Error: Failed to parse JSON file. {e}")
        return None

# Function to upload configuration file to the server
def upload_config_file(config_data, url):
    try:
        # Send the configuration data to the server
        response = requests.post(f'{url}/upload-config', json=config_data)

        if response.status_code == 200:
            print("Configurations uploaded successfully.")
        else:
            print(f"Failed to upload configurations: {response.status_code} {response.text}")
    except requests.exceptions.RequestException as e:
        print(f"Error: Failed to send request to the server. {e}")

# Main function for the CLI
def main():
    # Set up the argument parser
    parser = argparse.ArgumentParser(description="Admin CLI for Konorkestra")
    parser.add_argument('--upload-configs', type=str, help="Path to the configuration file to upload.")
    parser.add_argument('--url', type=str, required=True, help="URL of the server (e.g., http://localhost:5000).")

    args = parser.parse_args()

    # Ensure the user provided the config file path
    if not args.upload_configs:
        print("Error: You must provide a configuration file path using '--upload-configs'.")
        return

    # Read the configuration file
    config_data = read_config_file(args.upload_configs)
    if config_data is None:
        return  # If there was an error reading the file, stop further execution

    # Upload the configuration file to the server
    upload_config_file(config_data, args.url)

if __name__ == "__main__":
    main()

File: KonorkestraCLI/konorkestraCli.go
package KonorkestraCLI

import (
	"bytes"
	"encoding/json"
	"errors"
	"fmt"
	"github.com/jessevdk/go-flags"
	"io/ioutil"
	"net/http"
	"os"
)

// Function to read the configuration file
func readConfigFile(filePath string) (map[string]interface{}, error) {
	if _, err := os.Stat(filePath); os.IsNotExist(err) {
		return nil, fmt.Errorf("Error: The file %s does not exist.", filePath)
	}

	// Read and parse the JSON configuration file
	fileData, err := ioutil.ReadFile(filePath)
	if err != nil {
		return nil, fmt.Errorf("Error: Failed to read file. %s", err)
	}

	var configData map[string]interface{}
	if err := json.Unmarshal(fileData, &configData); err != nil {
		return nil, fmt.Errorf("Error: Failed to parse JSON file. %s", err)
	}

	return configData, nil
}

// Function to upload configuration file to the server
func uploadConfigFile(configData map[string]interface{}, url string) {
	// Convert configData to JSON
	configJSON, err := json.Marshal(configData)
	if err != nil {
		fmt.Printf("Error: Failed to convert config data to JSON. %s\n", err)
		return
	}

	// Send the configuration data to the server
	resp, err := http.Post(fmt.Sprintf("%s/upload-config", url), "application/json", bytes.NewBuffer(configJSON))
	if err != nil {
		fmt.Printf("Error: Failed to send request to the server. %s\n", err)
		return
	}
	defer resp.Body.Close()

	if resp.StatusCode == 200 {
		fmt.Println("Configurations uploaded successfully.")
	} else {
		fmt.Printf("Failed to upload configurations: %d %s\n", resp.StatusCode, resp.Status)
	}
}

type Args struct {
	UploadConfigs string `short:"u" long:"upload-configs" description:"Path to the configuration file to upload." required:"true"`
	URL           string `short:"r" long:"url" description:"URL of the server (e.g., http://localhost:5000)." required:"true"`
}

// Main function for the CLI
func main() {
	var args Args
	_, err := flags.Parse(&args)
	if err != nil {
		if errors.Is(err, flags.ErrHelp) {
			return
		}
		fmt.Println("Error:", err)
		return
	}

	// Read the configuration file
	configData, err := readConfigFile(args.UploadConfigs)
	if err != nil {
		fmt.Println(err)
		return
	}

	// Upload the configuration file to the server
	uploadConfigFile(configData, args.URL)
}
